dandi/dandiapi.py

```diff
@@ -1367,8 +1298,13 @@ def get_download_file_iter(
         """
         Returns a function that when called (optionally with an offset into the
         asset to start downloading at) returns a generator of chunks of the
-        asset
+        asset.
+
+        :raises ValueError: if the asset is not backed by a blob
         """
+        if self.asset_type is not AssetType.BLOB:
+            raise ValueError("Only blob assets can be downloaded directly")
```

Comment:
could we add here information about Asset (at least its type; and I know that we do have only 2 ATM so easy to exclude), e.g.
```suggestion
            raise ValueError(f"Only blob assets can be downloaded directly. The asset is of type {self.asset_type}")
```
? or may be Asset's `str` is "informative" and it could be just
```suggestion
            raise ValueError(f"Cannot download {self}: only blob assets can be downloaded directly")
```
or alike?

Reply:
Done: 34eaa01